node/modules/storageminer.go

```diff
@@ -113,14 +113,12 @@ func HandleRetrieval(host host.Host, lc fx.Lifecycle, m *retrieval.Miner) {
 	})
 }
 
-func HandleDeals(mctx helpers.MetricsCtx, lc fx.Lifecycle, host host.Host, h *deals.Provider) {
+func HandleDeals(mctx helpers.MetricsCtx, lc fx.Lifecycle, host host.Host, h storagemarket.StorageProvider) {
 	ctx := helpers.LifecycleCtx(mctx, lc)
 
 	lc.Append(fx.Hook{
 		OnStart: func(context.Context) error {
-			h.Run(ctx)
-			host.SetStreamHandler(deals.DealProtocolID, h.HandleStream)
-			host.SetStreamHandler(deals.AskProtocolID, h.HandleAskStream)
+			h.Run(ctx, host)
```

Comment:
I would pass in host to the constructor. And then let Run get called with just context. (still only register the handlers in run though)